src/infer_request.cc

```diff
@@ -1006,12 +1006,50 @@ InferenceRequest::Normalize()
     // If no input is marked as optional, then use exact match error message
     // for consistency / backward compatibility
     if ((size_t)model_config.input_size() == model_raw_->RequiredInputCount()) {
+      // Model config is master
+      // Original inputs is smaller
+      std::string missing_input_string = "[";
+      std::string original_input_string = "[";
+
+      int missing_input_cnt = 0;
+      for (size_t i = 0; i < (size_t)model_config.input_size(); ++i) {
+        const inference::ModelInput& input = model_config.input(i);
+        if (original_inputs_.find(input.name()) == original_inputs_.end()) {
+          missing_input_string =
+              missing_input_string + "'" + input.name() + "'" + ",";
+          missing_input_cnt++;
+        }
+      }
+      // Removes the extra ","
+      // Can be handled in a a more elegant way by creating a vector first of
+      // the inputs and then the string this way saves memory and overhead of
+      // creating a vector.
+      missing_input_string.pop_back();
+      missing_input_string = missing_input_string + "]";
+
+      for (const auto& pair : original_inputs_) {
+        original_input_string =
+            original_input_string + "'" + pair.first + "'" + ",";
+      }
+      // Removes the extra ","
+      // Can be handled in a a more elegant way by creating a vector first of
+      // the inputs and then the string this way saves memory and overhead of
+      // creating a vector.
```

Comment:
Don't need to duplicate the comment in both places, and I don't think you need the "can be handled in a more elegant way part". Just saying `Removes the extra ","` once is sufficient I think.